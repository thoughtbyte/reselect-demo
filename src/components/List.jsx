import React from "react";
import PropTypes from "prop-types";

const List = ({ list }) => {
  console.log("RENDERING List");

  return (
    <div className="card">
      <ul>{list.map(listItem => <li key={listItem}>{listItem}</li>)}</ul>
    </div>
  );
};

List.propTypes = {
  list: PropTypes.array.isRequired,
};

export default List;
