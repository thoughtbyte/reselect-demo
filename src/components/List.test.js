import React from "react";
import renderer from "react-test-renderer";

import List from "./List";

const props = {
  list: [
    "Clean Code",
    "The Pragmatic Programmer",
    "You Don't Know JS",
    "Eloquent JavaScript",
  ],
};

describe("List", () => {
  test("should match snapshot", () => {
    const list = renderer.create(<List {...props} />).toJSON();

    expect(list).toMatchSnapshot();
  });
});
