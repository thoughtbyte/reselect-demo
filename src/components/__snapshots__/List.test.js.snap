// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<div
  className="card"
>
  <ul>
    <li>
      Clean Code
    </li>
    <li>
      The Pragmatic Programmer
    </li>
    <li>
      You Don't Know JS
    </li>
    <li>
      Eloquent JavaScript
    </li>
  </ul>
</div>
`;
